### PR TITLE
Fix QuerySplitter from losing previous query parts

### LINKED ISSUE #13888: Selection conditions are lost in some cases with JOINs
### CrateDB version

- 5.2.3 / 5.2.5
- 5.1.4
- 5.0.3
- 4.8.4
- 4.7.3
- 4.2

### CrateDB setup information

single / multi node

### Problem description

filter conditions are lost in some cases with JOINs

### Steps to Reproduce

```sql
CREATE TABLE ta01 (a TEXT, b BOOLEAN, c BOOLEAN, p TEXT);
CREATE TABLE ta02 (p TEXT, pt TEXT);
EXPLAIN SELECT COUNT(*) FROM ta01 LEFT JOIN ta02 ON ta01.p = ta02.p
    WHERE NOT b and c;
```

### Actual Result

```
+--------------------------------------+
| EXPLAIN                              |
+--------------------------------------+
| HashAggregate[count(*)]              |
|   └ NestedLoopJoin[LEFT | (p = p)]   |
|     ├ Collect[doc.ta01 | [p] | c]    |
|     └ Collect[doc.ta02 | [p] | true] |
+--------------------------------------+
```

### Expected Result

```
+-----------------------------------------------+
| EXPLAIN                                       |
+-----------------------------------------------+
| HashAggregate[count(*)]                       |
|   └ NestedLoopJoin[LEFT | (p = p)]            |
|     ├ Collect[doc.ta01 | [p] | c AND (NOT b)] |
|     └ Collect[doc.ta02 | [p] | true]          |
+-----------------------------------------------+
```

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes #13888

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
retest this please, unrelated `MetadataTrackerITest` failures.

File: server/src/main/java/io/crate/analyze/relations/QuerySplitter.java [hunk 1]
```diff
@@ -136,7 +136,11 @@ public Void visitField(ScopedSymbol field, Context ctx) {
 
         @Override
         public Void visitReference(Reference ref, Context ctx) {
-            ctx.parts.put(Set.of(ref.ident().tableIdent()), ref);
+            Set<RelationName> relationName = Set.of(ref.ident().tableIdent());
+            Symbol prevQuery = ctx.parts.put(relationName, ref);
+            if (prevQuery != null) {
+                ctx.parts.put(relationName, AndOperator.of(prevQuery, ref));
+            }
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/analyze/relations/QuerySplitter.java:143]:
```suggestion
            ctx.parts.merge(Set.of(ref.ident().tableIdent()), ref, AndOperator::of);
```

**Review by mfussenegger — APPROVED** [review]

**Review by matriv — APPROVED** [review]
thank you!

File: server/src/main/java/io/crate/analyze/relations/QuerySplitter.java (hunk 1 above)
Comment (jeeminso) [reply server/src/main/java/io/crate/analyze/relations/QuerySplitter.java]: Thanks!

**Comment by jeeminso** [comment]
Thanks for reviewing @mfussenegger, @matriv. As I make the change from `Map.put()` to `Map.merge()`, noticed few other places (`visitField()` and `visitMatchPredicate()`) suffered from the same issue and fixed them. Also the release entry accordingly. Please have another quick look.

**Review by matriv — COMMENTED** [review]
thx! Can you please add test(s) for the other cases?
It would be great to show the fix and avoid breaking this in the future.

**Comment by jeeminso** [comment]
Thank you, added.

**Review by matriv — APPROVED** [review]

**Pull request merged**